content/en/blog/_posts/Creating AWS Volume Snapshots in Kubernetes Using the EBS CSI Driver

```diff
@@ -0,0 +1,651 @@
+
+**Note**: the volume snapshotting alpha release does not provide any data consistency guarantees. For example, if some new data was added to the database during the snapshotting process, it may be not reflected in the snapshot. Users should take corresponding actions like pausing their app or freezing a filesystem to ensure data consistency while taking volume snapshots. 
+
+Great! We have a MySQL volume snapshot at our disposal! Now let's do something as crazy as deleting your MySQL Deployment and the PVC with the data:
+
+```
+kubectl delete deployment mysql
+kubectl delete pvc ebs-claim
+```
+
+### Step #8: Restoring MySQL Data using the Volume Snapshot
+
+We deleted the database we created earlier, and all the data was gone. However, we have a volume snapshot, so don't worry! To provision a new AWS EBS volume pre-populated with the data from our snapshot, we must use a new `dataSource` field in the `PersistentVolumeClaim` . This field supports three parameters:
+
+- `name` — name of the `VolumeSnapshot` to use as a source
+- `kind` — must be `VolumeSnapshot`
+- `apiGroup` — must be `snapshot.storage.k8s.io`
+
+Also, note that the `storageClassName` should match a CSI StorageClass: `ebs-sc` 
+
+**Note**: the namespace of the `VolumeSnapshot` object is assumed to be the same as the namespace of the `PersistentVolumeClaim` object.
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-restore
+spec:
+  storageClassName: ebs-sc
+  dataSource:
+    name: mysql-snapshot
+    kind: VolumeSnapshot
+    apiGroup: snapshot.storage.k8s.io
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 4Gi
+```
+
+Create this PVC:
+
+```shell
+kubectl create -f pvc-snapshot.yaml 
+persistentvolumeclaim/pvc-restore created
+```
+
+Let's now create a brand new MySQL deployment with this new `pvc-restore` PVC attached:
+
+```yaml
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: mysql-clone
+spec:
+  selector:
+    matchLabels:
+      app: mysql-clone
+      version: "1"
+  strategy:
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 1
+    type: RollingUpdate
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: mysql-clone
+        version: "1"
+    spec:
+      containers:
+      - image: mysql:5.6
+        name: mysql
+        env:
+        - name: MYSQL_USER
+          value: root
+        - name: MYSQL_ROOT_PASSWORD
+          value: password
+        ports:
+        - containerPort: 3306
+        volumeMounts:
+        - name: mysql-persistent-storage
+          mountPath: /var/lib/mysql
+      volumes:
+      - name: mysql-persistent-storage
+        persistentVolumeClaim:
+          claimName: pvc-restore
+```
+
+Save this spec to the mysql-clone.yaml , and create new Deployment:
+
+```shell
+kubectl create -f mysql-clone.yaml
+deployment.apps/mysql-clone created
```

Comment:
The Kubernetes documentation style guide [uses](https://kubernetes.io/docs/contribute/style/style-guide/#separate-commands-from-output) one code block for the command and a second one for the expected output.